server/models/UserBan.ts

```diff
@@ -0,0 +1,24 @@
+import { Entity, ManyToOne, JoinColumn } from 'typeorm';
+import { BaseModel } from './BaseModel';
+import { User } from './User';
+import { Chapter } from './Chapter';
+
+@Entity({ name: 'user_bans' })
+export class UserBan extends BaseModel {
```

Comment:
Could we maybe instead of doing this create a `joinTable` https://typeorm.io/#/many-to-many-relations? Also the same for `EventSponsor` and `UserChapter`

Reply:
What do you think?

Reply:
yeah, we could. I was thinking of a situation where it doesn't extend `BaseModel` but I didn't test it first.

Reply:
https://typeorm.io/#/many-to-many-relations/many-to-many-relations-with-custom-properties :cry: , for them to have `createdAt` and `updatedAt` we need custom entities, so maybe we could create BaseModel that only has `createdAt` and `updatedAt` and add ID to all models that need it?

Reply:
I think what we currently have will also work since `UserBan` also needs an `id`.

Reply:
Ok, keep it like this then